src/main/scala/calc/Compiler.scala

```diff
@@ -70,10 +70,17 @@ object Compiler {
     // TODO
     implicit val pos = tree.pos
 
+    import irt.BinaryOp._
+    val binaryOpMap = Map("+" -> Double_+, "-" -> Double_-, "*" -> Double_*,
+      "/" -> Double_/)
```

Comment:
One small improvement: this map should be a `private val` in the enclosing object, so that it is not recomputed every time you call `compileExpr`.